src/cli.js

```diff
@@ -611,15 +611,24 @@ var exports = {
     }
 
     var filename;
+    var lintStdinFile;
 
     // There is an if(filename) check in the lint() function called below.
     // passing a filename of undefined is the same as calling the function
     // without a filename.  If there is no opts.filename, filename remains
     // undefined and lint() is effectively called with 4 parameters.
     if (opts.filename) {
-      filename = path.resolve(opts.filename);
+      filename = opts.filename;
+      var ignores = !opts.ignores ? loadIgnores({ cwd: opts.cwd }) :
+                                  opts.ignores.map(function(target) {
+                                    return path.resolve(target);
+                                  });
+      lintStdinFile = (opts.useStdin && !isIgnored(filename, ignores));
```

Comment:
This is a strange edge case (the user specifies `--filename` but does not explicitly enable reading from standard input), but your patch handles it correctly. Good thinking! Can you add a test to make sure we don't mess this up in the future?

Reply:
I've added a new test to cover this behaviour, please let me know what you think about it

Reply:
Looks good to me!